src/offliner-client.js

```diff
@@ -79,21 +79,78 @@
      * @return {Promise} A promise resolving if the installation success.
      */
     install: function () {
-      this._installMessageHandlers();
+      if (!('serviceWorker' in navigator)) {
+        return Promise.reject(new Error('serviceworkers-not-supported'));
+      }
+
       return navigator.serviceWorker.register(workerURL, {
         scope: root
-      });
+      }).then(function (registration) {
+        return this.connect().then(function () {
+          return registration;
+        });
+      }.bind(this));
     },
 
     /**
-     * If you are using offliner as a serviceworkerware middleware, instead
-     * of calling {{#crossLink OfflinerClient/install:method}}{{/crossLink}},
-     * call `connect()` to avoid registering the worker.
+     * Keeps the promise of connect.
+     *
+     * @property _connecting
+     * @type Promise
+     * @private
+     */
+    _connecting: null,
+
+    /**
+     * Connects the client with offliner allowing the client to control offliner
+     * and receive events.
      *
      * @method connect
+     * @return {Promise} A promise resolving once connection has been stablished
+     * with the worker and communication is possible.
      */
     connect: function () {
-      this._installMessageHandlers();
+      if (!this._connecting) { this._connecting = this._connect(); }
+      return this._connecting;
+    },
+
+    /**
+     * The actual implementation for {{#crossLink "connect:method"}}{{/crossLink}}
+     *
+     * @method _connect
+     * @return {Promise} A promise resolving once connection has been stablished
+     * with the worker and communication is possible.
+     * @private
+     */
+    _connect: function () {
+      if (!('serviceWorker' in navigator)) {
```

Comment:
Should we check this again here or is the check in install() enough? Presumably the developer won't try to connect if he can't install, right?
It doesn't really hurt checking again, so we can keep it if you think it could be needed.